tests/perf-system/benchmarks/benchmarks.py

```diff
@@ -0,0 +1,513 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import json
+import os
+import random
+import signal
+import stat
+import string
+import subprocess
+import sys
+import time
+from pathlib import Path
+from loguru import logger as LOG
+
+import paramiko
+
+
+def connect_to_remote_server(ip_address):
+    client = paramiko.SSHClient()
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    client.connect(ip_address)
+    return client
+
+
+def clean_up_cchost(nodes):
```

Comment:
In CI infra, we use context managers to make sure connections are shut down if the script terminates, and a combination of a pty to get a SIGHUP on disconnect (https://github.com/microsoft/CCF/blob/9e0e46c41a306d10dc128f3f3672ad16a58cc089/tests/infra/remote.py#L264) and a flag to terminate all children processes: https://github.com/microsoft/CCF/blob/9e0e46c41a306d10dc128f3f3672ad16a58cc089/tests/infra/remote.py#L38